test/e2e/utils/helpers.go

```diff
@@ -506,62 +506,96 @@ func GetMemberServiceSelector(scyllaClusterName string) labels.Selector {
 	}.AsSelector()
 }
 
-func GetScyllaHostsAndWaitForFullQuorum(ctx context.Context, client corev1client.CoreV1Interface, sc *scyllav1.ScyllaCluster) ([]string, error) {
-	scyllaClient, hosts, err := GetScyllaClient(ctx, client, sc)
-	if err != nil {
-		return nil, fmt.Errorf("can't get scylla client: %w", err)
-	}
-	defer scyllaClient.Close()
-
-	sortedHosts := make([]string, len(hosts))
-	copy(sortedHosts, hosts)
-	sort.Strings(sortedHosts)
-
-	// Wait for node status to propagate and reach consistency.
-	// This can take a while so let's set a large enough timeout to avoid flakes.
-	err = wait.PollImmediateWithContext(ctx, 1*time.Second, 5*time.Minute, func(ctx context.Context) (done bool, err error) {
-		allSeeAllAsUN := true
-		infoMessages := make([]string, 0, len(hosts))
-		var errs []error
-		for _, h := range sortedHosts {
-			s, err := scyllaClient.Status(ctx, h)
-			if err != nil {
-				return true, fmt.Errorf("can't get scylla status on node %q: %w", h, err)
-			}
+func GetScyllaHostsByDCAndWaitForFullQuorum(ctx context.Context, dcClientMap map[string]corev1client.CoreV1Interface, scs []*scyllav1.ScyllaCluster) (map[string][]string, error) {
+	allHosts := make(map[string][]string, 0)
+	var sortedAllHosts []string
 
-			sHosts := s.Hosts()
-			sort.Strings(sHosts)
-			if !reflect.DeepEqual(sHosts, sortedHosts) {
-				errs = append(errs, fmt.Errorf("node %q thinks the cluster consists of different nodes: %s", h, sHosts))
-			}
+	var errs []error
+	for _, sc := range scs {
+		client, ok := dcClientMap[sc.Spec.Datacenter.Name]
+		if !ok {
+			errs = append(errs, fmt.Errorf("client is missing for datacenter %q of ScyllaCluster %q", sc.Spec.Datacenter.Name, naming.ObjRef(sc)))
+			continue
+		}
+
+		hosts, err := GetHosts(ctx, client, sc)
+		if err != nil {
+			return nil, fmt.Errorf("can't get hosts for ScyllaCluster %q: %w", sc.Name, err)
+		}
+		allHosts[sc.Spec.Datacenter.Name] = hosts
+		sortedAllHosts = append(sortedAllHosts, hosts...)
+	}
+	err := errors.Join(errs...)
+	if err != nil {
+		return nil, err
+	}
 
-			downHosts := s.DownHosts()
-			infoMessages = append(infoMessages, fmt.Sprintf("Node %q, down: %q, up: %q", h, strings.Join(downHosts, "\n"), strings.Join(s.LiveHosts(), ",")))
+	sort.Strings(sortedAllHosts)
 
-			if len(downHosts) != 0 {
-				allSeeAllAsUN = false
-			}
+	for _, sc := range scs {
+		client, ok := dcClientMap[sc.Spec.Datacenter.Name]
+		if !ok {
+			errs = append(errs, fmt.Errorf("client is missing for datacenter %q of ScyllaCluster %q", sc.Spec.Datacenter.Name, naming.ObjRef(sc)))
+			continue
 		}
 
-		if !allSeeAllAsUN {
-			framework.Infof("ScyllaDB nodes have not reached status consistency yet. Statuses:\n%s", strings.Join(infoMessages, ","))
+		scyllaClient, hosts, err := GetScyllaClient(ctx, client, sc)
+		if err != nil {
+			return nil, fmt.Errorf("can't get scylla client: %w", err)
 		}
+		defer scyllaClient.Close()
```

Comment:
defer in for loop :(

please, wrap it in a function

Reply:
done